shark_turbine/kernel/wave/utils.py

```diff
@@ -108,3 +109,60 @@ def delinearize_index(index: IndexExpr, shape: list[int]) -> list[IndexExpr]:
             nd_index.append(sympy.floor(index / product) % size)
         product *= size
     return nd_index[::-1]
+
+
+def simplify_index(index: IndexExpr) -> IndexExpr:
+    """
+    Simplifies the index by applying the following bindings:
+        - MMA acc_index bindings so the index of the MMA node is the acc_index.
+    """
+    mapping = {MMA_LHS: 0, MMA_RHS: 0, MMA_ACC: 1}
+    return index.subs(mapping)
+
+
+def get_mma_dimensional_mapping(trace: CapturedTrace) -> dict[IndexSymbol, int]:
+    """
+    Given a trace, determine the MMA dimensional mapping for all the
+    MMA operations in the graph. For example, if we have
+        acc = tkw.mma(a_reg, b_reg, acc)
+    where a_reg has shape UxV, b has shape SxV and acc has shape UxS,
+    we map U to the MMA M dimension (0), S to the MMA N dimension (1) and
+    V to the MMA K dimension (2).
+    """
+
+    def is_mma(node: fx.Node) -> bool:
+        custom = get_custom(node)
+        return isinstance(custom, MMA)
+
+    mma_nodes = trace.walk(is_mma)
+    mapping: dict[IndexSymbol, int] = {}
+    for node in mma_nodes:
+        custom: MMA = get_custom(node)
+        m, n = custom.acc_type.symbolic_shape[-2:]
+        mapping[m] = 0
+        mapping[n] = 1
+        k = [
+            dim
+            for dim in custom.lhs_type.symbolic_shape
```

Comment:
Maybe NIT:  Personal preference to express it as sets operations
```
k = list(set(custom.lhs_type.symbolic_shape).intersection(custom.rhs_type.symbolic_shape) - set(custom.acc_type.symbolic_shape))[0]
```

but this works too :)

Reply:
> if dim in custom.rhs_type.symbolic_shape

Is custom.rhs_type.symbolic_shape already a set? if it is then probably even better to use the first comment set operations.  :)

Reply:
I dont think it is a set, but I think we can write this more concisely using sets. I will change.